docs/coding-conventions.md

```diff
@@ -0,0 +1,89 @@
+# Coding conventions
+
+This document outlines a collection of guidelines, style suggestions, and tips
+for writing code in the KubeVirt project. It is partially based on [Kubernetes
+Coding Conventions](https://github.com/kubernetes/community/blob/master/contributors/guide/coding-conventions.md).
+
+- [Go Code Review Comments](https://github.com/golang/go/wiki/CodeReviewComments)
+- [Effective Go](https://golang.org/doc/effective_go.html)
+- Know and
+  avoid [Go landmines](https://gist.github.com/lavalamp/4bd23295a9f32706a48f)
+- Inline variable assignments when possible
+    - For example inline `err` checks
+- Use early returns to avoid unnecessarily deeply nested if/else blocks
+- Use switch-cases to avoid long if/else if/else blocks
+- Avoid named return values
+    - Define variables in the function body instead
+- Avoid closures when they do not capture any meaningful state
+    - For example capturing the KubeVirt client
+    - Use simple functions instead
+- Use structs and receiver methods to keep state
+  - Avoid use of global variables
+- Isolate code where possible
+    - Use interfaces instead of bare structs
+- Use constants to avoid repetition of hardcoded values
+    - Keep them private if they are used only in a single package
+    - Make the exported if they are used in multiple places
+- Declare empty slices with the var syntax
+    - Pay attention when serializing data,
+      see [Declaring Empty Slices](https://go.dev/wiki/CodeReviewComments#declaring-empty-slices)
+- Avoid use of `fmt.Sprintf` when possible
```

Comment:
seems a bit too broad

Reply:
Yes, I had these specific examples below in mind when writing this. I'll try to make this more specific.